cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -172,16 +172,27 @@ private void cleanupQueue() {
       // Queue head might have already been removed. So instead of calling poll, we call remove.
       if (launchingQueue.remove(programRunId)) {
         LOG.info("Removing request with runId {} due to expired retention time.", programRunId);
-        emitMetrics(Constants.Metrics.FlowControl.WORKFLOWS_LAUNCHING_COUNT, launchingQueue.size());
+        emitMetrics(Constants.Metrics.FlowControl.LAUNCHING_COUNT, launchingQueue.size());
       }
     }
   }
 
-  private int getWorkflowsRunningCount() {
+  /**
+   * Returns the total number of programs in running state.
+   * The count includes batch (i.e., {@link ProgramType#WORKFLOW}),
+   * streaming (i.e., {@link ProgramType#SPARK}) with no parent and
+   * replication (i.e., {@link ProgramType#WORKER}) jobs.
+   *
+   * @return
+   */
+  private int getProgramsRunningCount() {
     List<ProgramRuntimeService.RuntimeInfo> list = runtimeService
-      .listAll(ProgramType.WORKFLOW);
+      .listAll(ProgramType.WORKFLOW, ProgramType.WORKER, ProgramType.SPARK, ProgramType.MAPREDUCE);
```

Comment:
are we to count SPARK program ? (usually a SPARK program is included in WORKFLOW program)

Reply:
We need to count spark programs for streaming jobs as streaming job have Spark type. Note that we are not double counting (see the above comment reply)

Reply:
got it, thanks for the information :)